### PR TITLE
Hide empty lists

### PR DESCRIPTION
**Pull request by cybervinit**
Hides the empty lists (i.e. lists without any cards or lists with only `HOLD` cards)

### CODE REVIEW

File: src/services/api.js [hunk 1]
```diff
@@ -73,6 +73,12 @@ const filterByMention = async(list) => {
     .then(filtered => filtered);
 };
 
+const filterNonEmptySections = async(list) => {
```
Comment (schoettler) [inline src/services/api.js]: Why is this an asynchronous function?

File: src/services/api.js [hunk 2]
```diff
@@ -73,6 +73,12 @@ const filterByMention = async(list) => {
...
+  return list.filter((el, i) => {
```
Comment (schoettler) [inline src/services/api.js]: nit: rename el to `card`

File: src/services/api.js [hunk 3]
```diff
@@ -73,6 +73,12 @@ const filterByMention = async(list) => {
...
+    return !(((i + 1) === list.length && el.isSectionHeading) || (list[i].isSectionHeading && list[i + 1].isSectionHeading));
```
Comment (schoettler) [inline src/services/api.js]: Ah neat, you're checking for consecutive Section headings, or a last empty section heading. 

We may be able to make this easier to read

File: src/services/api.js (hunk 1 above)
Comment (cybervinit) [reply src/services/api.js]: It did not need to be. removed `async`

File: src/services/api.js (hunk 2 above)
Comment (cybervinit) [reply src/services/api.js]: done

File: src/services/api.js (hunk 3 above)
Comment (cybervinit) [reply src/services/api.js]: simplified condition and added `if` to improve readability

**Pull request merged**